overlord/snapstate/handlers.go

```diff
@@ -1098,8 +1098,12 @@ func (m *SnapManager) doCopySnapData(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	st.Lock()
+	opts := GetSnapDirOptions(st)
```

Comment:
ultimately if we do migration one snap at a time we will have to track both the feature flag and the per-snap state of whether it's using the old dir or the new hidden dir

Reply:
my comment is probably more relevant for ~/Snap that what we do here, though it depends on the migration we will implement